duckduckgo/duckduckgo-privacy-extension
# Improved broken site reporting

**Reviewer:** @MariagraziaAlastra 

**CC:** @jdorweiler 

## Description:
This PR streamlines the process for broken site reporting, and makes the reports more useful to us. It's important to note that this is all strictly opt-in, and it is totally up to the user if they would like to share any data at all about broken sites. The main changes are:
- The form is now totally contained within the popup, and triggered upon disabling privacy protection.
- The free form text field has been replaced with a categorization select menu, so that we may better classify breakage at scale. The free form text field produced a lot of noise that didn't help us to diagnose issues and fix them.
- If a user opts in to sharing extra data about a broken site with us, a few additional parameters will be added to the form. These include the domains of trackers that were blocked on the page, whether an https upgrade was performed on the site, and whether any surrogates were active on the page. The url of the site in question will also be attached to the form, with params and fragments removed to protect privacy.

## Steps to test this PR:
1. Open background page network tab
2. Visit any normal domain with this installed
3. Open popup and click whitelist toggle. An ept.off pixel should fire and a small form should appear under the toggle asking if the page is broken.
4. Click no in that form. It should thank you for your feedback and reload the page.
5. Remove site from whitelist by toggling privacy protection in the pop up. Check the network tab, an ept.on pixel should have fired that doesn't include the site url.
6. Repeat 2-4, this time hit yes.
7. Breakage categorization form should appear, with the submit button disabled.
8. Select a category from the dropdown. Submit button should enable.
9. Click submit. Thank you message should fade in and page should automatically reload in a couple seconds.
10. Check the network tab of the background page. An epbf pixel should have fired with the site url (minus fragments and params), trackers blocked, https upgrade status, category of breakage, surrogates, and atb. Trackers blocked and surrogates are comma separated values, and the site url should be URI encoded properly.
11. On the same site, toggle privacy protection back on. Check the network tab again, an ept.on pixel should have fired with the site domain and path attached.
12. Remove site from whitelist, open the popup and click report broken site in either the hamburger menu or the main area. Breakage form should appear. Fill it out and hit submit. This time you should see the thank you message, then the form disappear but no page reload or popup close.
13. Get to breakage form again, and X out of it. If you got to breakage form from toggling privacy protection, the form should close and the page should reload. Otherwise, the form should close but the page shouldn't reload.
14. Visit special site such as new tab page. Open the popup and click the hamburger menu. The report broken site option should be hidden. We limited this form to only submitting reports on the current page, so it doesn't make sense to allow users to submit reports about local pages.


TODO: finish unit tests, internal pr to register pixel.

## Automated tests:

###### Reviewer Checklist:


###### PR Author Checklist:
